File: src/main/java/com/develetter/develetter/userfilter/entity/UserFilter.java
package com.develetter.develetter.userfilter.entity;

import com.develetter.develetter.global.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@Entity
@Getter
@SuperBuilder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Table(name = "user_filter")
public class UserFilter extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Embedded
    private JobPostingKeyword jobpostingKeywords;  // 임베디드 타입

    @Column(name = "blog_keywords", nullable = false, columnDefinition = "TEXT")
    private String blogKeywords;
}

File: backend/src/main/java/com/develetter/develetter/userfilter/entity/UserFilter.kt
package com.develetter.develetter.userfilter.entity

import com.develetter.develetter.global.entity.BaseEntity
import com.develetter.develetter.userfilter.dto.UserFilterReqDto
import jakarta.persistence.*

@Entity
@Table(name = "user_filter")
class UserFilter(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    var id: Long? = null,

    @Column(name = "user_id", nullable = false)
    var userId: Long? = null,

    @Embedded
    var jobpostingKeywords: JobPostingKeyword? = null, // 임베디드 타입

    @Column(name = "blog_keywords", columnDefinition = "TEXT")
    var blogKeywords: String? = null

) : BaseEntity() {
    fun update(dto: UserFilterReqDto) {
        this.jobpostingKeywords = JobPostingKeyword(
            dto.jobNames,
            dto.locationNames,
            dto.jobTypeNames,
            dto.industryNames,
            dto.educationLevelNames
        )
        this.blogKeywords = dto.blogKeywords
    }
}
